src/plugins/embeddable/public/lib/embeddables/embeddable.tsx

```diff
@@ -48,15 +59,33 @@ export abstract class Embeddable<
   // to update input when the parent changes.
   private parentSubscription?: Rx.Subscription;
 
+  private storageSubscription?: Rx.Subscription;
+
   // TODO: Rename to destroyed.
   private destoyed: boolean = false;
 
-  private __actionStorage?: EmbeddableActionStorage;
-  public get actionStorage(): EmbeddableActionStorage {
-    return this.__actionStorage || (this.__actionStorage = new EmbeddableActionStorage(this));
+  private storage = new EmbeddableActionStorage(this);
+
+  private cachedDynamicActions?: UiActionsDynamicActionManager;
+  public get dynamicActions(): UiActionsDynamicActionManager | undefined {
+    if (!this.params.uiActions) return undefined;
+    if (!this.cachedDynamicActions) {
+      this.cachedDynamicActions = new UiActionsDynamicActionManager({
+        isCompatible: async ({ embeddable }: any) => embeddable.runtimeId === this.runtimeId,
```

Comment:
> remove `any` type.

Done.

> I suspect that setting this isCompatible function here is going to prevent devs from implementing their own isCompatible fns.  We can address in a follow up - I added it as a topic to the tech debt.

Yes, but it is being taken care of, we have this in mind for the future work for making Drilldown UI usable outside of dashboard. There was `doExecute` method (which of course can be renamed to `isCompatible`) on `DrilldownDefinition` interface, which I thought we would merge here with this `isCompatible`.

```
{
  isCompatibe: ({ embeddable }) => doExecute() && embeddable.runtimeId === this.runtimeId,
}
```

This is one of those details because of which it is nice to have a separate registration flow for drilldowns, because we have to modify the uderlying `isCompatible` function of the actions, so it does not come as a surprise to users.